File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<vector<int>> arr(nums.size()+1);
        unordered_map<int,int>mp;
        for(int i=0;i<nums.size();i++){
            if(mp.find(nums[i])!=mp.end()){
                mp[nums[i]]++;
            }
            else{
                mp[nums[i]]=1;
            }
            vector<int> a;
            arr[i] = a;
        }

        vector<int> a;
        arr[nums.size()-1] = a;
        for(auto t:mp){
            arr[t.second].push_back(t.first);
        }

        vector<int> ans;

        int number=nums.size();
        while(k!=0){
            if(arr[number].size()!=0){

                for(int i=0;i<arr[number].size();i++){
                    ans.push_back(arr[number][i]);
                    k--;
                }

                // while(k!=0 and arr[number].size()!=0){
                //     k--;
                //     ans.push_back(arr[number]);
                // }
            }
            number--;
        }
        return ans;

    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-3.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        st = []
        ans = 0
        for k in tokens:
            if k=="+":
                a = st.pop()
                b= st.pop()

                ans = a+b
                st.append(ans)
                # st.append(ans)
            elif k=="-":
                a = st.pop()
                b= st.pop()

                ans = b-a
                st.append(ans)
            elif k=="*":
                a = st.pop()
                b= st.pop()

                ans = a*b
                st.append(ans)

            elif k=="/":
                a = st.pop()
                b= st.pop()

                ans = int(b/a)
                st.append(ans)
            else:
                print(int(k))
                st.append(int(k))
            print(st)
        return st[0]

        
        